src/jdk.internal.vm.ci/share/classes/jdk/vm/ci/hotspot/HotSpotJVMCIRuntime.java

```diff
@@ -926,22 +926,16 @@ JavaType lookupTypeInternal(String name, HotSpotResolvedObjectType accessingType
 
     /**
      * Gets the {@code jobject} value wrapped by {@code peerObject}. The returned "naked" value is
-     * only valid as long as {@code peerObject} is valid. Note that the latter may be shorter than
-     * the lifetime of {@code peerObject}. As such, this method should only be used to pass an
-     * object parameter across a JNI call from the JVMCI shared library to HotSpot. This method must
-     * only be called from within the JVMCI shared library.
-     *
+     * a JNI local reference, which is valid for the duration of a JVMCI shared library call. This
```

Comment:
"which is valid for the duration of a JVMCI shared library call"

This is a bit ambiguous for a reader not fully aware of the dual JNI environments in play. One may think the call to this method is a "JVMCI shared library call". Can you express this some other way that doesn't make it sound like the return value is invalid?

Reply:
Fixed by d10f81847f27dfe413ace2521aec2f76647f4071